### PR TITLE
feat(java): relocate SpannerFeature from java-core

### PR DESCRIPTION
**Pull request by mpeddada1**
This PR moves SpannerFeature from java-core to Spanner. Once this PR is merged, we can remove https://github.com/googleapis/java-core/blob/main/native-image-support/src/main/java/com/google/cloud/nativeimage/features/clients/SpannerFeature.java  

Note that the Kokoro Native Image job don't run any tests with native image support at the moment. This will be resolved with https://github.com/googleapis/java-spanner/pull/1812

### CODE REVIEW

File: google-cloud-spanner/pom.xml [hunk 1]
```diff
@@ -104,21 +114,19 @@
...
-            <spanner.testenv.config.class>${spanner.testenv.config.class}</spanner.testenv.config.class>
-            <spanner.testenv.instance>${spanner.testenv.instance}</spanner.testenv.instance>
-            <spanner.gce.config.project_id>${spanner.gce.config.project_id}</spanner.gce.config.project_id>
-            <spanner.testenv.kms_key.name>${spanner.testenv.kms_key.name}</spanner.testenv.kms_key.name>
-          </systemPropertyVariables>
-          <forkedProcessTimeoutInSeconds>3000</forkedProcessTimeoutInSeconds>
-        </configuration>
-      </plugin>
+<!--      <plugin>-->
+<!--        <groupId>org.apache.maven.plugins</groupId>-->
```
Comment (ansh0l) [inline google-cloud-spanner/pom.xml:118]: Should we just remove these commented config? Or do you intend to revise the PR later @mpeddada1 ?

Comment (mpeddada1) [reply google-cloud-spanner/pom.xml:118]: Thanks @ansh0l! I kept these in for testing purposes but yes, we remove these before merging the PR.

Comment (mpeddada1) [reply google-cloud-spanner/pom.xml]: All set. I've removed this comment from the pom.xml.

**Comment by ansh0l** [comment]
Closing since native image changes have been released with https://github.com/googleapis/java-spanner/pull/1878

**Pull request closed without merging**